Rx433.cpp

```diff
@@ -62,63 +62,58 @@ bool Handle(std::vector<Pulse> buf) {
 
 void ICACHE_RAM_ATTR rxISR() {
   static int last_changed = 0, sync_pulse_us = 0;
+  static int last_state = 0;
   static std::vector<rx433::Pulse> pulse_stream;
-  bool pulse_stream_done = false;
-  if (pulse_stream_queue.size() > 10) return;
   int now = micros();
+  int state = digitalRead(rxPin_);
+  int delta_us = now - last_changed;
+
+  // Ignore very short glitches that we missed.
+  if (state == last_state) return;
+  last_state = state;
+
+  // Streamed glich filter
+  // Ignore any pulse (1 or 0) shorter than kGlitchUs.
+  if (!pulse_stream.empty() && delta_us < kGlitchUs) {
+    last_changed -= pulse_stream.back().delta_us;
+    pulse_stream.pop_back();
+    return;
+  }
+  last_changed = now;
+
+  // Queue is full, drop pulse
+  if (pulse_stream_queue.size() > 10) return;
```

Comment:
I would move this up (to save CPU). It should be first thing after static variables

Reply:
I moved it down as I think the other checks (same state, glitch) will trigger more and thus save more CPU. These ought to be more time critical as well.